### DISCUSSION TITLE
Creating custom tags for docker-compose files.

### DISCUSSION THREAD

**Discussion by mshima** [eemeli/yaml#561] [Q&A]
Docker compose files supports tags that does not change the yaml property type.
Example:
```
services:
  app:
    ports: !override
      - "8443:443" 
```

Would be nice to parse/stringify it using something like:
```
const flags = Symbol.for('yaml-flags');
const ports = ['8443:443'];
ports[flags] = ['override'].
const content = stringify({ services: { app: { ports } } });
// yaml example above

const parsed = parse(content);
console.log(parsed.services.app.ports[flags]);
// ['override']
```

Is this possible?

Reference:
 - https://github.com/compose-spec/compose-spec/blob/5c18e329d5a15a15e4b636ed093b256b96615e33/13-merge.md#replace-value
 - https://github.com/compose-spec/compose-spec/blob/5c18e329d5a15a15e4b636ed093b256b96615e33/13-merge.md#reset-value

**Comment by eemeli** [comment]
This is a bit reminiscent of the idea proposed in #294, of effectively extending Object, Array, and other values with YAML metadata. That's pretty tricky. Working with custom tags like `!override` is currently provided via the Document API:

```js
import { parseDocument } from 'yaml'

const doc = parseDocument(`
services:
  app:
    ports: !override
      - "8443:443"
`)
const seq = doc.getIn(['services', 'app', 'ports'])
seq.tag // '!override'
```

**Reply by mshima** [reply]
Document api uses custom classes correct?
So it's not possible to our use case.

**Reply by eemeli** [reply]
Note that with your proposed use of `Symbol.for('yaml-flags')`, you are also defining a custom object interface.

**Comment by mshima** [comment]
With current customTag api it's not possible?

**Reply by eemeli** [reply]
With that you can specify how `!override` works, but you need to specify a behaviour for each tag specifically.

**Reply by mshima** [reply]
I've tried a customTag like:
```
{
  tag: '!override',
  identify: value => {
    console.log('identify', value);
    return value?.override === true;
  },
  resolve: value => {
    console.log('resolve', value);
    value.override = true;
    return value;
  },
},
```

Didn't work, identify and resolve are not called at deserialization.
